src/sysupdate/sysupdate.c

```diff
@@ -744,6 +745,24 @@ static int context_make_online(Context **ret, const char *node) {
         return 0;
 }
 
+static int context_on_acquire_progress(const Transfer *t, const Instance *inst, unsigned percentage, void *userdata) {
+        const Context *c = ASSERT_PTR(userdata);
+        size_t i, n = c->n_transfers;
+        unsigned overall;
+
+        for (i = 0; i < n; i++)
+                if (c->transfers[i] == t)
+                        break;
+        assert(i < n); /* We should have found the index */
+
+        overall = (percentage / n) + (i * (100 / n));
```

Comment:
this is integer multiplication, hence please multiply by 100 first, and divide then, to retain accuracy. Also maybe do calculation in uint64_t to avoid any risk of this overflow.